packages/settings-library/src/settings_library/utils_service.py

```diff
@@ -2,14 +2,24 @@
 
 
 """
+from enum import Enum, auto
+from typing import Optional
+
 from pydantic.networks import AnyUrl
+from pydantic.types import SecretStr
 
 from .basic_types import PortInt
 
 DEFAULT_AIOHTTP_PORT: PortInt = 8080
 DEFAULT_FASTAPI_PORT: PortInt = 8000
 
 
+class URLPart(Enum):
+    EXCLUDE = auto()
+    OPTIONAL = auto()
+    REQUIRED = auto()
```

Comment:
just wondering about this auto method.. just checked in the documentation. how does that work if we add another value? it has to be appended right? at least if the value goes somewhere?

Reply:
it just adds a different value.
the way i use it is: if the enum is not going to be stored , then I just care that they are different and leave it to ``auto`` to choose them. If it is persisted, then I am the one settings the values explicitly